java/src/jmri/jmrit/logixng/tools/swing/LogixNGEditor.java

```diff
@@ -457,6 +457,12 @@ void newConditionalNGPressed(ActionEvent e) {
                 _systemName.setText(InstanceManager.getDefault(ConditionalNG_Manager.class).getAutoSystemName());
             }
 
+            ConditionalNG cn = _curLogixNG.getConditionalNGByUserName(_addUserName.getText());
```

Comment:
This looks for duplicates within the current LogixNG.  The create looks at all of the conditionals.

Note:  In 2017 I refactored Logix/Conditionals to make conditionals children of its related Logix which removed them from the bean management domain, i.e _tsys and _tuser.  Conditionals are still named beans, but only accessible via the parent Logix.

Reply:
Thanks. Fixed in last commit.

Yes, it would probably be a good idea to refactor LogixNG/ConditionalNG to have the ConditionalNGs only accessible from their LogixNG. But I don't have the time to do it now so if I should do it, it will have to wait until later.

If you want to refactor LogixNG/ConditionalNG to have the ConditionalNGs only accessible from their LogixNG, feel free to do it. I'm fine with that.

If you want to do it, do you prefer to do it now or after next production release? If you prefer to do it now, I will stand with you in that. Bob Jacobsen has probably the ability to build the next test release around May 20 or 21. And it's easier to do it now when it's almost only you and me that's using LogixNG at this point. So it might be a good idea to do it now. And if so, I think it would be a good idea to ask Bob to postpone the next test release until this is done.

Reply:
Thanks.

Reply:
Bob Jacobsen has planned to build the next test release on Wednesday May 12. It seems reasonable that we ask him to postpone it to May 20 or 21, with a note that it might take longer. We can refer to [this discussion](https://github.com/JMRI/JMRI/pull/9779#discussion_r629674369) and that we want to refactor LogixNG before it gets into the production version.

I'm going to bed now but will be online again in about 10 - 12 hours.

Reply:
I can go either way.  I think that is Bob's choice.

I lean towards .5 this week.  That provides 5 weeks until .6.

Reply:
@dsand47 I did a try on this today and it seems a lot more easier to do than I first thought. I remove the ConditionalNG_Manager, remove system name from ConditionalNG, ensure that old panel files still works and ensure that the tests still works. I think I will have it working in a couple of days.

@bobjacobsen You can build .5 now and I think the refactorisation will be ready to merge within a week.

Reply:
> @dsand47 I did a try on this today and it seems a lot more easier to do than I first thought. I remove the ConditionalNG_Manager, remove system name from ConditionalNG, ensure that old panel files still works and ensure that the tests still works. I think I will have it working in a couple of days.

It sounds like you are taking a very different approach than I am.  I have everything in place except Logix Import.

My approach is that ConditionalNGs are not registered in the InstanceManager.  They are added to a map in the parent LogixNG.

Logix uses the _conditionalMap along with _conditionalSystemNames.  LogixNG uses _conditionalNG_Entries which appears to combine the two Logix maps.  There is a residual _conditionalNGMap that does not appear to be used very much.

Reply:
Lets go with your approach. It sounds good.

Reply:
@bobjacobsen 
The refactorization of LogixNG / ConditionalNG is now merged into master with #9784 so you can create the next test version when you have time for it.

There is some other PRs that might be reasonable to include as well in this test version, but it's your choice.

#9792 updates the LogixNG documentation so it would be good to include.

#9783 / #9785 seems to be much wanted by users. #9785 includes #9783.

#9789 solves a problem for a user who wants to create a script.

#9782 adds two new decoder files.

Reply:
A bit confused.  Is the goal to have 4.23.next from master with changes, or from a specific branch with only LNG changes?  Happy to do it either way.

Reply:
I understand the confusion. The goal was to have only the LogixNG changes in the next test release. But since there are a couple of PRs that would be good to have in the next production release, I think it would be better to create the next test release from master.

I have merged those PRs that was ready to merge (approved, 24h passed, CI passed).

The current status:
#9789 is waiting for CI to pass. Once clear, I will merge it.
#9785 is waiting for someone to review it. Once it's approved, it's ready to be merged.
#9798 is waiting for 24 hours to pass.
#9797 is waiting for 24 hours to pass.